mobile/src/src/components/Menu/Profile/EditComponent.jsx

```diff
@@ -0,0 +1,691 @@
+          <Picker.Item
+            key={i}
+            label={tempChoice[i]}
+            value={i}
+            style={styles.modalOptions}
+          />
+        );
+      }
+    }
+
+    return temp;
+  }
+
+  const onPickerChange = (itemValue, itemIndex, type) => {
+    var zero = 999;
+    if (itemValue == 0) {
+      zero = 0;
+    }
+    if (type == "Pronoun") {
+      setZeroPronoun(zero);
+      setPronoun(itemValue);
+    } else if (type == "Region") {
+      setZeroReg(zero);
+      setRegion(itemValue);
+    } else if (type == "Program") {
+      setZeroPro(zero);
+      setProgram(itemValue);
+    } else if (type == "Current Term") {
+      setZeroTerm(zero);
+      setTerm(itemValue);
+    }
+  };
+
+  function getSpecificModal(type) {
+    return (
+      <View style={styles.centeredView}>
+        <View style={styles.modalView}>
+          <Text style={styles.modalText}>Pick a {type} Option!</Text>
+          <View style={styles.modalButton}>
+            <Picker
+              selectedValue={
+                type == "Pronoun"
+                  ? pronoun
+                  : type == "Region"
+                  ? reg
+                  : type == "Program"
+                  ? program
+                  : type == "Current Term"
+                  ? term
+                  : ""
+              }
+              style={styles.modalPicker}
+              onValueChange={(itemValue, itemIndex) =>
+                onPickerChange(itemValue, itemIndex, type)
+              }
+            >
+              <Picker.Item label="--Options--" style={styles.modalOptions} />
+              {getOptions(type)}
+            </Picker>
+          </View>
+          <TouchableHighlight
+            style={styles.openButton}
+            onPress={() => {
+              type == "Pronoun"
+                ? setVisiblePronoun(!visiblePronoun)
+                : type == "Region"
+                ? setVisibleReg(!visibleReg)
+                : type == "Program"
+                ? setVisiblePro(!visiblePro)
+                : type == "Current Term"
+                ? setVisibleTerm(!visibleTerm)
+                : "";
+            }}
+          >
+            <Text style={styles.textStyle}>Close</Text>
+          </TouchableHighlight>
+        </View>
+      </View>
+    );
+  }
+
+  const onDateChange = (event, selectedValue) => {
+    setDate(selectedValue);
+    const currYear = selectedValue.getFullYear().toString();
+    const currMonth =
+      selectedValue.getMonth() < 10
+        ? ("0" + (selectedValue.getMonth() + 1)).toString().slice(-2)
+        : (selectedValue.getMonth() + 1).toString().slice(-2);
+    const currDay =
+      selectedValue.getDate() < 10
+        ? ("0" + selectedValue.getDate()).toString().slice(-2)
+        : selectedValue.getDate().toString().slice(-2);
+    const currDate = currYear + "-" + currMonth + "-" + currDay;
+    console.log(currDate);
+    setBirth(currDate);
+  };
+
+  function getBirthModal(type) {
```

Comment:
@shashankanandkannan I used the DateTimePicker here, I have it in a modal bc on ios it displays like a scrolling thing, but on android it should display like a calendar.